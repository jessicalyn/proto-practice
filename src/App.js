import React, { Component } from 'react';
import { protoPractice } from './protoPractice'

class App extends Component {
  constructor() {
    super();

    this.state = {
      allQuestions: protoPractice,
      studyGuideQuestions: [],
      practiceActive: false
    }
  }

  // componentDidMount() {
  //   this.setState({allQuestions: protoPractice})
  // }
  startPractice() {

  }

  render() {
    return (
      <div className="App">
        <h1>ProtoPractice</h1>
        <h2>Build your knowledge of array, object, and string prototypes!</h2>
        <h3>For each question you'll see an image of a prototype problem with the prototype name missing. Choose the correct prototype from the answers listed below.</h3>
        <h3>Answers missed with be added to your study guide for futher practice.</h3>
        <button onClick={this.startPractice}>Start Practicing</button>
        <button>Reset Study Guide</button>
        <Question allQuestions/>
      </div>
    );
  }
}

export default App;
