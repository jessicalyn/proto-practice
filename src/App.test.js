import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme'

const mockStudyGuideQuestions = [{
  answerList: ["push", "includes", "shift"],
  category: "Array Prototype",
  correctAnswer: "shift",
  docs: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift",
  id: "a72913de-36c6-11e9-b210-d663bd873d93",
  img: "images/shift.png"
  },
  {
  answerList: ["shift", "sort", "splice"],
  category: "Array Prototype",
  correctAnswer: "splice",
  docs: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice",
  id: "a72919d8-36c6-11e9-b210-d663bd873d93",
  img: "images/splice.png"
}];

const mockQuestion = [{
  answerList: ["push", "includes", "shift"],
  category: "Array Prototype",
  correctAnswer: "shift",
  docs: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift",
  id: "a72913de-36c6-11e9-b210-d663bd873d93",
  img: "images/shift.png",
}];

describe('App', () => {
  const wrapper = shallow(
    <App />
  );

  it('should match a snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have proper default states', () => {
    expect(wrapper.state()).toEqual({"activeQuestions": [], "allQuestions": [], "currentIndex": 0, "practiceRound": "start", "studyGuideQuestions": []})
  })

  it('should start practice round when invoked', () => {
    expect(wrapper.state("practiceRound")).toEqual("start")
    wrapper.instance().startPractice()
    expect(wrapper.state("practiceRound")).toEqual("flashcards")
  })

  it('should move to next available question when invoked', () => {
    wrapper.state.activeQuestions = [30]
    wrapper.state.currentIndex = 0

    wrapper.instance().nextQuestion()
    expect(wrapper.state("currentIndex")).toEqual(1)
  })

  // it('should change practice round state when no available questions', () => {
  //   wrapper.setState({
  //     practiceRound: "flashcards", 
  //     roundIndex: 29, 
  //     currentIndex: 29,
  //   });   

  //   wrapper.instance().nextQuestion();

  //   expect(wrapper.state("practiceRound")).toEqual("end")
  // })

  it('should add incorrect questions to study guide array when invoked', () => {
    expect(wrapper.state("studyGuideQuestions").length).toEqual(0)
    wrapper.instance().addToStudyGuide(mockQuestion)
    expect(wrapper.state("studyGuideQuestions").length).toEqual(1)
  })

  it('should update state for study guide array when invoked', () => {
    wrapper.setState({
      activeQuestions: [30], 
      studyGuideQuestions: mockStudyGuideQuestions, 
      currentIndex: 29,
      practiceRound: "end"
    });

    wrapper.instance().useStudyGuide();

    expect(wrapper.state("activeQuestions")).toEqual(mockStudyGuideQuestions)
    expect(wrapper.state("studyGuideQuestions")).toEqual([])
    expect(wrapper.state("currentIndex")).toEqual(0)
    expect(wrapper.state("practiceRound")).toEqual("flashcards")
  })

  it('should reset state to start when invoked', () => {
    wrapper.setState({
      studyGuideQuestions: mockStudyGuideQuestions, 
      practiceRound: "flashcards", 
      currentIndex: 10,
      activeQuestions: mockQuestion
    });

    wrapper.instance().resetGame();

    expect(wrapper.state("studyGuideQuestions")).toEqual([])
    expect(wrapper.state("practiceRound")).toEqual("start")
    expect(wrapper.state("currentIndex")).toEqual(0)
    expect(wrapper.state("activeQuestions")).toEqual([])
  })


})

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
