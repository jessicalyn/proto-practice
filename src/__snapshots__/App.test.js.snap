// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot with all data passed in 1`] = `
<div
  className="App"
>
  <h1>
    ProtoPractice
  </h1>
  <h2>
    Build your knowledge of array, object, and string prototypes!
  </h2>
  <Start
    startPractice={[Function]}
  />
</div>
`;
